graph/GuidelinesGraph.md

```diff
@@ -0,0 +1,441 @@
+|-----------------------------------------------------------------------------------------------------------------------|
+| :heavy_check_mark: **MUST** return an error property with a child code property in all error responses.                 | 
+| :heavy_check_mark: **MUST** return a 403 Forbidden error when insufficient scopes are present in the auth token.        | 
+| :heavy_check_mark: **MUST** return a 429 Too many requests error when the caller has exceeded throttling limits.        | 
+| :ballot_box_with_check: **SHOULD** return a 404 Not found error if a 403 would result in information disclosure. |
+
+For a complete mapping of error codes to HTTP statuses you can refer to the
+[rfc7231 (ietf.org)](https://datatracker.ietf.org/doc/html/rfc7231#section-6).
+
+## API contract and non-backward compatible changes
+
+Microsoft Graph definition of breaking changes is based on the [Microsoft REST
+API
+Guidelines](https://github.com/microsoft/api-guidelines/blob/graph/Guidelines.md#123-definition-of-a-breaking-change).
+In general, making all but additive changes to the API contract for existing elements is
+considered breaking. Adding new elements is allowed and not considered a
+breaking change.
+
+\*\* Non-breaking changes:\*\*
+
+
+-   Addition of properties that are nullable or have a default value
+-   Addition of a member to an evolvable enumeration 
+-   Removal, rename, or change to the type of an annotation   
+-   Changes to the order of properties
+-   Changes to the length or format of opaque strings, such as resource IDs
+-   Addition or removal of an annotation OpenType="true" 
+
+\*\* Breaking changes:\*\*
+
+-   Changes to the URL or fundamental request/response associated with a
+    resource
+-   Removal, rename, or change to an incompatible type of a declared property
+-   Removal or rename of APIs or API parameters
+-   Addition of a required request header
+-   Addition of a EnumType members for non-evolvable enumerations
+-   Addition of a Nullable="false" properties to existing types
+-   Addition of a Nullable="false" parameters to existing actions and functions
+-    Changes to top-level error codes
+-    Introduction of server-side pagination to existing collections
+-    Changes to the default order of collection elements
+-    Significant changes to the performance of APIs such as increased latency, rate limits or concurrency.
+
+
+
+### Versioning and Deprecation
+As the market and technology evolves your APIs will require modifications in this case you must avoid breaking changes and add new resources and features incrementally. If that is not possible then you must version elements of your APIs.
+Microsoft Graph allows versioning of elements including entities and properties. The versioning process goes along with deprecation and as soon as you introduce a new element update the previous version needs to follow the deprecation process.
```

Comment:
Wording reads a bit awkward.  Maybe something like:

Microsoft Graph allows versioning of elements including entities and properties. Versioning involves adding a new, uniquely named version of the element and marking the old version as deprecated.

In some cases, there will be a natural new name for the element. In other cases, where the original name is still the most descriptive, the suffix _v2 can be added to the original name to make it unique.